### ISSUE TITLE
After adding cache settings in advancedsettings.xml, videos play for a bit then stop

### ISSUE THREAD

**Issue by ckeech** [xbmc/xbmc#15496]
### Describe the bug
Here is a clear and concise description of what the problem is:

After adding cache settings in advancedsettings.xml, videos play for a bit then stop

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Videos are expected to play as normal but with the cache bar filling up.

## Actual Behavior

Videos stop playing, and exit back to the previous menu. Kodi stays responsive.

## Possible Fix

Removing advancedsettings.xml fixes the issue. However then it removes the ability to have cache.

### To Reproduce

1. Create advancedsettings.xml with settings in https://kodi.wiki/view/HOW-TO:Modify_the_video_cache
2. Play video

### Debuglog

The debuglog can be found here: https://paste.kodi.tv/veyegopadu.kodi

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

- I use powerline adapters for my home network
- Works quite well, but occasionally (say once or twice in every second episode) a video will pause and buffer for a bit then resume
- The fix was to increase the cache settings, I’ve been using
`<advancedsettings>
  <cache>
    <buffermode>1</buffermode>
    <memorysize>139460608</memorysize>
    <readfactor>20</readfactor>
  </cache>
</advancedsettings>`

For ages and it has fixed the issue. Looks like the network can drop out for a little bit but with the cache settings it has a little while to catch up and therefore doesn’t have any visible pauses.
- Updated my LibreElec to 9.0 (Leia)
- I had issues with videos crashing.
- I started from a clean image and imported all my media again. Narrowed it down to when I add the advanced settings.xml with the cache is when it starts. Symptom is, play a video, it starts, plays about  2 seconds and closes. No error and Kodi is responsive afterwards.

- I tried adjusting <readfactor> values, no difference.
- I removed <readfactor> and left just <memorysize> - no difference
- I adjusted <memorysize>. Halved it, then divided it by 10. No difference.
- Removing advancedsettings.xml fixes the issue however brings back my original issue.

- Searched “cache” and “cache crash” in the forums and github. Seems like no one else has experienced this.

- Does it happen on all files? Well short answer yes. It seems really small files take longer to die.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [x] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: LibreELEC 9.0.0
 - Kodi version: 18.0

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by arnova** [comment]
I suspect your device is running out of memory. Try to remove the <memorysize>-tag and see how it goes.

**Comment by ckeech** [comment]
Yes, removing the memorysize-tag would stop the issue, for me it's the same as deleting advancedsettings.xml as it's the only setting I've got in there now. I wondered if it was running out of memory too, but there's a few things that's odd with that.
1 - I have the exact same advancedsettings.xml with memorysize setting before I updated to Kodi v18 and it was working fine. On a Pi3 and 2. Then after v18 both have the same issue wih the video crashing.
2 - Even if for some reason v18 uses more RAM, I halved the memorysize and then tried dividing by 10 and it still does the same thing.

Looking at Kodi v18 on my Pi 3 when it's idle is using 241MB of memory out of 747MB available. So the setting I had of 139460608 should use 532MB RAM according to that Kodi article. This does look like too much, so maybe Kodi v18 is using more RAM. Either way, when I change it to 79741304, that should use 304MB of RAM leaving 202MB RAM free. Video still crashes after about half a second of playing.
